tests/integration/tier0/config-file/test_config_file.py

```diff
@@ -0,0 +1,100 @@
+import os
+
+
+def create_files(paths):
+    for path, content in paths:
+        f = open(os.path.expanduser(path), "w")
+        f.write(content)
+        f.close()
+        os.chmod(os.path.expanduser(path), 0o600)
+
+
+def remove_files(paths):
+    for path, _ in paths:
+        os.remove(os.path.expanduser(path))
+
+
+def test_user_path_custom_filename(convert2rhel):
+    paths = (("~/.convert2rhel_custom.ini", "[subscription_manager]\nactivation_key = config_activationkey"),)
```

Comment:
I don't like the fact that `paths` contains the path to the file and the content as well. I would probably named it something like `config`.  Also using the `namedtuple` could be nice here and would make the code more readable I believe.

I would create a global level object:
`Config = namedtuple('Config', 'path content')`

And then in each test function I would do something like this:
```
config = Config(path="~/.convert2rhel_custom.ini", content="[subscription_manager]\nactivation_key = config_activationkey")
```
You can then easily pass the object to the `create_files` function and accress the variables like this:
`config.path` and `config.content` which is much nicer.

More on that: https://realpython.com/python-namedtuple/